### PR TITLE
Support theme meta w3c standard

### PR DESCRIPTION
**Pull request by skjnldsv**
https://html.spec.whatwg.org/multipage/semantics.html#meta-color-scheme

image

### CODE REVIEW

**Comment by szaimen** [comment]
With this I suppose we can remove https://github.com/nextcloud/server/blob/cfc1c21a80c90f8668846ed528f7b4355cbb6c24/apps/theming/lib/Controller/ThemingController.php#L376 again?

**Comment by skjnldsv** [comment]
> With this I suppose we can remove
> 
> https://github.com/nextcloud/server/blob/cfc1c21a80c90f8668846ed528f7b4355cbb6c24/apps/theming/lib/Controller/ThemingController.php#L376
> 
> again? (I must admit using the primary color as background-color is a misuse)

That's not related to this pr, is it?

**Comment by szaimen** [comment]
> That's not related to this pr, is it?

Not really, no. Could be done in a follow-up I suppose...
image
image

**Comment by skjnldsv** [comment]
> > That's not related to this pr, is it?
> 
> Not really, no. Could be done in a follow-up I suppose... image image

But it's also kinda related to the feature, sorry I misunderstood :)
I'll see that after the weekend :see_no_evil:

**Review by artonge — APPROVED** [review]

File: apps/theming/lib/ITheme.php [hunk 1]
```diff
@@ -70,6 +70,14 @@ public function getEnableLabel(): string;
 	 */
 	public function getDescription(): string;
 
+	/**
+	 * Get the meta attribute matching the theme
+	 * e.g. https://html.spec.whatwg.org/multipage/semantics.html#meta-color-scheme
+	 * @return string
```
Comment (CarlSchwan) [inline apps/theming/lib/ITheme.php:76]:
```suggestion
	 * @return array{?name: string, ?content: string}
```

**Review by CarlSchwan — APPROVED** [review]
Nice feature :)

File: apps/theming/lib/ITheme.php (hunk 1 above)
Comment (skjnldsv) [reply apps/theming/lib/ITheme.php:76]: I will probably allow multiple arrays of meta, to also provide support for the `background_color` per theme

**Comment by skjnldsv** [comment]
> @skjnldsv sorry for asking but is this still needed? If yes, would you mind rebasing and resolving conflicts? Otherwise I could try if you want :)

Feel free to dive at it :)

**Comment by szaimen** [comment]
> > @skjnldsv sorry for asking but is this still needed? If yes, would you mind rebasing and resolving conflicts? Otherwise I could try if you want :)
> 
> Feel free to dive at it :)

Rebased and fixed conflicts! :)

**Comment by ShGKme** [comment]
This PR changes `focus-visible` styles for me.

image

I have a dark system theme and a light Nextcloud theme. With this PR, I have
```html
<meta name="color-scheme" content="light dark">
```

Is `light dark` an expected value? Why `dark` if I have the light theme?

With `light` or `light light` values - outline is the same as it was before this PR:

image

**Pull request merged**